test-utils/src/commonMain/kotlin/FlowTest.kt

```diff
@@ -111,6 +112,18 @@ abstract class BaseFlowTest<TC : TestContext, T, F : Flow<T>>(val scope: Corouti
 
     private lateinit var testChannel: Channel<Pair<TestBlock<TC, T>, EmptyCompletableDeferred>>
 
+    @AfterTest
+    override fun afterTest() {
+        runBlocking {
+            val cookie = cookie
+            withContext(Dispatchers.Main.immediate) {
+                val testContext = contextMap.remove(cookie)
+                testContext?.dispose()
```

Comment:
Can you extract to a method? (this will also automatically make cookie a parameter and thus local scope, no longer needing `val cookie = cookie`)